### DISCUSSION TITLE
How to increase the text sizes of menu items and dialogue labels?

### DISCUSSION THREAD

**Discussion by HubKing** [LMMS/lmms#6702] [Support request] [answered]
The text size on the main menu bar or the button sizes seemed all right, but the text sizes of menu items and dialogue labels were too small. I have searched Google and tried QT_SCREEN_SCALE_FACTORS and QT_AUTO_SCREEN_SCALE_FACTOR. The problem is that whilst they did scale up the UI, the relative smallness remains, which makes the small parts are still quite small, and the normal parts now look stupidly huge. 

Is there a way to adjust the scales of specific elements of the UI, such as labels on dialogues, not the whole UI at once?

![image](https://user-images.githubusercontent.com/10900489/236186653-f89cd2eb-7481-403d-85b2-330385dfca34.png)

![image](https://user-images.githubusercontent.com/10900489/236187264-38a88100-d914-4e75-979f-fe220b396321.png)

**Comment by michaelgregorius** [comment, marked as answer]
Hi @HubKing,

both of the problems should be fixed if the pull requests #6779 and #6787 are merged.

Update: Both the PRs and a few additional ones like #7185 #7493 have been merged.

Please see [here](https://github.com/LMMS/lmms/issues/2510#issuecomment-1650212452) and [here](https://github.com/LMMS/lmms/issues/2510#issuecomment-1656591470) for images and videos that demonstrate the fixes of the problems that you have mentioned.

## Technical details
There are currently several places in the code where hard coded sizes are used. These are often defined in units of pixels which might have worked in the Full HD era but not anymore in the HiDPI era. These places in the code will need to be adjusted one by one which will be quite some work.

The goal should be to make LMMS work without needing to fall back on things like setting `QT_SCREEN_SCALE_FACTORS` and `QT_AUTO_SCREEN_SCALE_FACTOR`. LMMS is the only application I know that needs these "crutches" and as is demonstrated by your screen shots they don't even work well.
